Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,520 @@
+            {
+                if (player == null)
+                    continue;
+
+                player.OnClientExitLobby();
+            }
+        }
+
+        #region server handlers
+
+        public override void OnServerConnect(NetworkConnection conn)
+        {
+            if (numPlayers >= maxConnections)
+            {
+                conn.Disconnect();
+                return;
+            }
+
+            // cannot join game in progress
+            if (SceneManager.GetActiveScene().name != LobbyScene)
+            {
+                conn.Disconnect();
+                return;
+            }
+
+            base.OnServerConnect(conn);
+            OnLobbyServerConnect(conn);
+        }
+
+        public override void OnServerDisconnect(NetworkConnection conn)
+        {
+            if (conn.playerController != null)
+            {
+                NetworkLobbyPlayer player = conn.playerController.GetComponent<NetworkLobbyPlayer>();
+
+                if (player != null)
+                    lobbySlots.Remove(player);
+            }
+
+            allPlayersReady = false;
+
+            foreach (NetworkLobbyPlayer p in lobbySlots)
+            {
+                if (p != null)
+                    p.GetComponent<NetworkLobbyPlayer>().ReadyToBegin = false;
+            }
+
+            base.OnServerDisconnect(conn);
+
+            if (SceneManager.GetActiveScene().name == LobbyScene)
+                RecalculateLobbyPlayerIndices();
+
+            OnLobbyServerDisconnect(conn);
+        }
+
+        public override void OnServerAddPlayer(NetworkConnection conn)
+        {
+            if (SceneManager.GetActiveScene().name != LobbyScene) return;
+
+            if (lobbySlots.Count == maxConnections) return;
+
+            allPlayersReady = false;
+
+            if (LogFilter.Debug) Debug.LogFormat("NetworkLobbyManager:OnServerAddPlayer playerPrefab:{0}", lobbyPlayerPrefab.name);
+
+            GameObject newLobbyGameObject = OnLobbyServerCreateLobbyPlayer(conn);
+            if (newLobbyGameObject == null)
+                newLobbyGameObject = (GameObject)Instantiate(lobbyPlayerPrefab.gameObject, Vector3.zero, Quaternion.identity);
+
+            NetworkLobbyPlayer newLobbyPlayer = newLobbyGameObject.GetComponent<NetworkLobbyPlayer>();
+
+            lobbySlots.Add(newLobbyPlayer);
+
+            RecalculateLobbyPlayerIndices();
+
+            NetworkServer.AddPlayerForConnection(conn, newLobbyGameObject);
+        }
+
+        void RecalculateLobbyPlayerIndices()
+        {
+            if (lobbySlots.Count > 0)
+            {
+                for (int i = 0; i < lobbySlots.Count; i++)
+                {
+                    lobbySlots[i].Index = i;
+                }
+            }
+        }
+
+        public override void ServerChangeScene(string sceneName)
+        {
+            if (sceneName == LobbyScene)
+            {
+                Application.targetFrameRate = 10;
```

Comment:
this seems a bit arbitrary, why is this here?
why not let networkmanager handle it?

Reply:
Because there's no need to have the CPU screaming at 99% for a UI scene, and I think this predated you implementing targetFramerate management, or at least my being aware you had.